File: api/src/main/java/cn/hirrao/javaee/mapper/DrugAlertMapper.java
package cn.hirrao.javaee.mapper;

import cn.hirrao.javaee.entity.DrugAlert;
import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import org.apache.ibatis.annotations.Delete;
import org.apache.ibatis.annotations.Insert;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Update;

import java.sql.Date;
import java.time.LocalDate;
import java.time.LocalTime;

@Mapper
public interface DrugAlertMapper extends BaseMapper<DrugAlert> {
    @Insert("INSERT INTO drugAlert VALUES (#{alertId},#{uid},#{drugId}, #{alertTime},#{eatTime})")
    void insertDrugAlert(long alertId, long uid, long drugId, LocalTime alertTime, LocalDate eatTime);

    @Update("UPDATE drug SET isActive=#{isActive} WHERE alertId=#{alertId}")
    void updateDrugAlertIsActiveById(long alertId, int isActive);

    @Update("UPDATE drugAlert SET eatTime=#{eatTime} WHERE alertId=#{alertId}")
    void updateDrugAlertEatTimeById(long alertId, Date eatTime);

    @Delete("DELETE FROM drugAlert WHERE alertId=#{alertId} AND uid=#{uid} AND drugId=#{drugId}")
    void deleteDrugAlertById(long alertId, long uid, long drugId);
}

File: api/src/main/kotlin/com/hirrao/javaee/mapper/DrugAlertMapper.kt
package com.hirrao.javaee.mapper

import com.hirrao.javaee.entity.DrugAlert
import com.baomidou.mybatisplus.core.mapper.BaseMapper
import org.apache.ibatis.annotations.Delete
import org.apache.ibatis.annotations.Insert
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Update
import java.time.LocalDate
import java.time.LocalTime

@Mapper
interface DrugAlertMapper : BaseMapper<DrugAlert?> {
    @Insert("INSERT INTO drugAlert VALUES (#{alertId},#{uid},#{drugId}, #{alertTime},#{eatTime})")
    fun insertDrugAlert(alertId: Long, uid: Int, drugId: Long, alertTime: LocalTime?, eatTime: LocalDate?)

    @Update("UPDATE drug SET isActive=#{isActive} WHERE alertId=#{alertId}")
    fun updateDrugAlertIsActiveById(alertId: Long, isActive: Int)

    @Update("UPDATE drugAlert SET eatTime=#{eatTime} WHERE alertId=#{alertId}")
    fun updateDrugAlertEatTimeById(alertId: Long, eatTime: LocalDate?)

    @Delete("DELETE FROM drugAlert WHERE alertId=#{alertId} AND uid=#{uid} AND drugId=#{drugId}")
    fun deleteDrugAlertById(alertId: Long, uid: Int, drugId: Long)
}
